### PR TITLE
Refactors src/utils.ts

### LINKED ISSUE #293: Web Server: Refactor utils.ts
The functions contained within `src/utils.ts` need to b reviewed, refactored and moved to their ideal location.

### PR DESCRIPTION
**Pull request by sorsaffari**
## What is the goal of this PR?
Cleans up and moves the functions in `src/utils.ts` to a more appropriate location. 

## What are the changes implemented in this PR?
closes #293 
- moves Github API-related functions from `src/utils.ts` to `src/controllers/auth/utils.ts`
- bugfix: corrects the value of `config/dashboardPath`
- bugfix: changes route type for auth endpoints
- bugfix: binds auth route handlers to the controller object (to scope `this` correctly)
- bugfix: configures `.env` _before_ checking for missing env variables

### CODE REVIEW

File: service/web-server/src/controllers/auth/utils.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    userAccessToken: undefined |string;
+
+    setUserAccessToken: () => Promise<void>;
+    getUserId: () => Promise<string>;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getGraknLabsMembers: () => Promise<any>;
+    revokeAccess: () => Promise<void>;
+}
+
+export function GithubClient(oauthCode: string = ''): IGithubClient {
```
Comment (marco-scoppetta) [inline service/web-server/src/controllers/auth/utils.ts]: only minor comment on this function name, if we call it `GithubClient` I would expect it to be a `Constructor`, and therefore to use it like `new GithubClient()`, whereas this function instead returns an object literal.
So maybe something like `getGtihubClient()`, or `initGithubClient()`?
Just a suggestion, no need to change it now

Comment (sorsaffari) [reply service/web-server/src/controllers/auth/utils.ts]: do you think the same should apply in these places:

https://github.com/graknlabs/benchmark/blob/master/service/web-server/src/server.ts#L15

https://github.com/graknlabs/benchmark/blob/master/service/web-server/src/routes/auth.ts#L9

Comment (marco-scoppetta) [reply service/web-server/src/controllers/auth/utils.ts]: Yes, I missed those, I think in general it's good to distinguish between constructor function and "factory" function, so nobody gets confused with the use of `new` (which when used incorrectly leads to very funny bugs)

Comment (sorsaffari) [reply service/web-server/src/controllers/auth/utils.ts]: Got it. I'll apply this change for `GithubClient` in this PR and for the other ones in the next PR

**Pull request merged**